src/app/views/provisioning/subscription.controller.coffee

```diff
@@ -19,12 +18,14 @@ angular.module 'mnoEnterpriseAngular'
     MnoeProvisioning.fetchSubscription($stateParams.id).then(
       (response) ->
         vm.subscription = response
-        if vm.subscription.custom_data?
+        unless _.isEmpty(vm.subscription.custom_data)
           vm.model = vm.subscription.custom_data
-          MnoeMarketplace.findProduct(id: vm.subscription.product_id).then(
-            (response) ->
-              vm.schema = if response.custom_schema then JSON.parse(response.custom_schema) else {}
-              vm.form = if response.asf_options then JSON.parse(response.asf_options) else ["*"]
+
+          MnoeMarketplace.fetchCustomSchema(vm.subscription.product_id).then((response) ->
+            # Some products have custom schemas, whereas others do not.
+            resp = JSON.parse(response)
+            vm.schema = if resp?.json_schema then resp.json_schema else resp
+            vm.form = if resp?.asf_options then resp.asf_options else ["*"]
```

Comment:
can these be refactored this way?
```
vm.schema = resp?.json_schema || resp
vm.form = resp?.asf_options || ["*"]
```

Reply:
It is more idiomatic in coffeescript to use an if-then-else one liner: https://coffeescript.org/#conditionals